### ISSUE TITLE
Silence the startup command printout in Integrated Terminal

### ISSUE THREAD

**Issue by mymikemiller** [microsoft/vscode#100522]
When I run a Dart app in the Integrated Terminal, it prints out a very verbose startup command, e.g.:

`cd /Users/mikem/myproject; /usr/local/Cellar/dart/2.8.2/libexec/bin/dart --enable-vm-service=0 --pause_isolates_on_start=true --write-service-info=file:///var/folders/2n/vn59_vz152vg107rrs53ympc0000gn/T/dart-vm-service-ee74.json -DSILENT_OBSERVATORY=true --enable-asserts bin/main.dart`

This happens for Javascript too:

`mikem@MBP node_sample %  /usr/local/bin/node --inspect-brk=6018 scripts/simple.js 
Debugger listening on ws://127.0.0.1:6018/453abe68-1da3-4862-8ba4-71d8e3dfd8d6
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.`

These printouts pollute the scrollback and aren't very useful imho. Could we get an option to silence this output, so the first thing I see in the Integrated Terminal is one of the print statements actually in my code?

Note that this doesn't happen when using the Debug Console. There, I only see this printout (which also would be nice to silence, but at least it's less verbose): 

`Connecting to VM Service at ws://127.0.0.1:58210/4yciVJ1NsWE=/w`

**Comment by weinand** [comment]
@Tyriar is there a good way to executed a command in the integrated terminal but turning off echoing?

**Comment by weinand** [comment]
@mymikemiller if the debug console works for you, why are you not using it instead of the integrated terminal?

**Comment by ArturoDent** [comment]
You can turn off echoing commands in bash for example.

`set +x`

in your `C:\Program Files\Git\etc\bash.bashrc`

There is also a `verbose +/-y` switch in some shells.  See https://stackoverflow.com/questions/2853803/how-to-echo-shell-commands-as-they-are-executed

I don't know if either of those would affect the startup commands that you are seeing?

**Comment by mymikemiller** [comment]
@weinand The debug console doesn't accept interactive input, but also it doesn't allow me to position the cursor absolutely, which is required for a [library](https://pub.dev/packages/console) I'm using to print out nice progress bars.

**Comment by mymikemiller** [comment]
@ArturoDent I don't think this is a bash thing (I use zsh, but same thing). When I run `dart bin/main.dart`, my app runs without echoing any command. So this seems to be a VSCode issue.

**Comment by mymikemiller** [comment]
Btw, I get the same output with bash as my default terminal as I do with zsh:

`bash-3.2$  cd /Users/mikem/myproject ; /usr/local/Cellar/dart/2.8.2/libexec/bin/dart --enable-vm-service=0 --pause_isolates_on_start=true --write-service-info=file:///var/folders/2n/vn59_vz152vg107rrs53ympc0000gn/T/dart-vm-service-f583.json -DSILENT_OBSERVATORY=true --enable-asserts bin/main.dart `

**Comment by mymikemiller** [comment]
@ArturoDent Switching to Bash and adding `set +x` to the top of my .bashrc does *not* get rid of the printout. It does stop my bashrc from printing out any of its commands, which are printed if changed to `set -x`.

**Comment by weinand** [comment]
@mymikemiller are you saying that when you are typing `dart bin/main.dart` in zsh, you are not seeing what you are typing?

**Comment by mymikemiller** [comment]
@weinand correct. 

In a normal terminal:

<img width="379" alt="Screen Shot 2020-06-23 at 2 09 25 PM" src="https://user-images.githubusercontent.com/2063229/85455709-689d1d80-b55b-11ea-8285-f6b953b5b29d.png">

In VSCode's integrated terminal when debugging:

<img width="434" alt="Screen Shot 2020-06-23 at 2 10 40 PM" src="https://user-images.githubusercontent.com/2063229/85455841-95513500-b55b-11ea-8b05-eee13688c16a.png">

**Comment by DanTup** [comment]
FWIW, when debugging here - the app is being spawned with the DAP's `RunInTerminalRequest` request, it's not being run explicitly by manually the command in the terminal.

**Comment by weinand** [comment]
@mymikemiller your zsh echos the command you are typing. The `dart bin/main.dart` corresponds to the command in your second screen shot.

<img width="381" alt="2020-06-24_16-29-53" src="https://user-images.githubusercontent.com/1898161/85575592-f9492b80-b637-11ea-8515-fdc5c9ab6858.png">

The only difference is: VS Code creates a much longer command from data received from the dart extension because it has to make sure that the command is run in the correct directory and the environment variables are correctly set up. This result in a longer command.

**Comment by mymikemiller** [comment]
@weinand It's not my zsh printing it, though. In the first screenshot, the text is there from me typing in the command and pressing enter. It doesn't print out the command again.

**Comment by mymikemiller** [comment]
The issue also occurs when bash is my default shell:

<img width="554" alt="Screen Shot 2020-06-25 at 9 50 09 AM" src="https://user-images.githubusercontent.com/2063229/85753653-aae46d00-b6c9-11ea-8bd2-91ac8443bdcd.png">

**Comment by weinand** [comment]
@mymikemiller in the second screenshot the text is there from VS Code typing in the command and pressing enter.

So the behavior is identically. You typing or VS Code typing makes no difference.
You don't see the text twice in the second screenshot, right?

The only difference is that VS Code has more to type.

**Comment by mymikemiller** [comment]
This is a feature request, though, not an issue. (I think I filed it as such)

From the initial post: "Could we get an option to silence this output, so the first thing I see in the Integrated Terminal is one of the print statements actually in my code?"

I think that's a valid request for a feature.

**Comment by weinand** [comment]
Sorry, but you called it issue:
> When I run dart bin/main.dart, my app runs without echoing any command. So this seems to be a VSCode **issue**.

I tried to explain that this is not a VS Code issue because VS Code is not "printing" out anything.

VS Code is just "typing" a long command that starts with "cd /Users/....; bin/main.dart" into the Integrated Terminal and the shell in that terminal echos the command in the same way it is echoing when you type manually.

I will open this as a feature request.

**Comment by DanTup** [comment]
In current Insiders build, this is *really* noisy for NodeJS now:

<img width="1406" alt="Screenshot 2020-07-09 at 13 30 06" src="https://user-images.githubusercontent.com/1078012/87040286-64723080-c1e8-11ea-8fd4-f351375bde31.png">

**Comment by testforstephen** [comment]
+1

We also received feedback from multiple users complaining that the startup command printed into the integrated terminal is too long when running Java application in terminal.
![image](https://user-images.githubusercontent.com/14052197/104554019-8f957b00-5676-11eb-85cb-3fa374c3cf8e.png)

**Comment by testforstephen** [comment]
Currently, if the start command is long, it is wrapped into multiple lines by the terminal. Users cannot quickly find the real starting point of the program output. This is the main pain point.

If the integrate terminal allows disabling the line wrapping, that will mitigate this issue too.

I found there is an existing feature request to support disabling line wrapping in integrated terminal.
